docs/adr/0007-azure-pipeline-security-model.md

```diff
@@ -0,0 +1,70 @@
+# Azure pipeline security model
+
+## Status
+
+Proposed
+
+
+## Context
+
+This ADR defines an approach for a least-privilege Azure DevOps Pipeline that supports the automated deployment, teardown and re-deployment of a complete Marain instance.
+
+The challenge here is to balance the varied benefits of having a self-sufficient, repeatable pipeline with the need for adequate security controls to minimise the risk of the pipeline becoming the source of a security incident.
+
+Such a deployment pipeline needs to be able to perform the following actions:
+
+1. Create resources in the target subscription
+1. Query AAD for existing applications & service principals
+1. Perform ARM role assignments in the target subscription
```

Comment:
What do you mean by ARM role assignments? To me, this implies things like "contributor" and "owner" roles for resource groups or even entire subs.

Ah...or is this for scenarios such as access control to storage accounts, which end up working through the ARM RBAC mechanism?